lib/ld-eventsource/events.rb

```diff
@@ -11,6 +11,10 @@ module SSE
   #     if there were multiple `data:` lines, they are concatenated with newlines
   # @!attribute id
   #   @return [String] the string that appeared after `id:` in the stream if any, or nil
+  # @!attribute last_event_id
+  #   @return [String] the most recent non-empty `id:` value that has been seen in an
+  #     event from this stream; this differs from the `id` property in that it retains
+  #     the same value in subsequent events if they do not provide their own `id:`
   #
-  StreamEvent = Struct.new(:type, :data, :id)
+  StreamEvent = Struct.new(:type, :data, :id, :last_event_id)
```

Comment:
This is not a breaking change because in a constructor produced by `Struct`, the arguments are all optional. So, any existing code that called `StreamEvent.new` with only three arguments would still work.